src/main/java/org/jabref/model/strings/LatexToUnicodeAdapter.java

```diff
@@ -5,23 +5,56 @@
 import java.util.regex.Pattern;
 
 import com.github.tomtung.latex2unicode.LaTeX2Unicode;
+import fastparse.core.Parsed;
 
 /**
  * Adapter class for the latex2unicode lib. This is an alternative to our LatexToUnicode class
  */
 public class LatexToUnicodeAdapter {
 
-    private static Pattern underscoreMatcher = Pattern.compile("_(?!\\{)");
+    private static final Pattern UNDERSCORE_MATCHER = Pattern.compile("_(?!\\{)");
 
-    private static String replacementChar = "\uFFFD";
+    private static final String REPLACEMENT_CHAR = "\uFFFD";
 
-    private static Pattern underscorePlaceholderMatcher = Pattern.compile(replacementChar);
+    private static final Pattern UNDERSCORE_PLACEHOLDER_MATCHER = Pattern.compile(REPLACEMENT_CHAR);
 
+    /**
+     * Attempts to resolve all LaTeX in the String.
+     *
+     * @param inField a String containing LaTeX
+     * @return a String with LaTeX resolved into Unicode, or the original String if the LaTeX could not be parsed
+     */
     public static String format(String inField) {
         Objects.requireNonNull(inField);
 
-        String toFormat = underscoreMatcher.matcher(inField).replaceAll(replacementChar);
-        toFormat = Normalizer.normalize(LaTeX2Unicode.convert(toFormat), Normalizer.Form.NFC);
-        return underscorePlaceholderMatcher.matcher(toFormat).replaceAll("_");
+        try {
+            return parse(inField).orElse(Normalizer.normalize(inField, Normalizer.Form.NFC));
+        } catch (IllegalArgumentException ignored) {
+            return Normalizer.normalize(inField, Normalizer.Form.NFC);
+        }
+    }
+
+    /**
+     * Attempts to resolve all LaTeX in the String.
+     *
+     * @param inField a String containing LaTeX
+     * @return a String with LaTeX resolved into Unicode
+     * @throws IllegalArgumentException if the LaTeX could not be parsed
+     */
+    public static Optional<String> parse(String inField) throws IllegalArgumentException {
+        Objects.requireNonNull(inField);
+        String toFormat = UNDERSCORE_MATCHER.matcher(inField).replaceAll(REPLACEMENT_CHAR);
+        try {
+            var parsingResult = LaTeX2Unicode.parse(toFormat);
+            if (parsingResult instanceof Parsed.Success) {
+                String text = parsingResult.get().value();
+                toFormat = Normalizer.normalize(text, Normalizer.Form.NFC);
+                return Optional.of(UNDERSCORE_PLACEHOLDER_MATCHER.matcher(toFormat).replaceAll("_"));
+            } else {
+ return Optional.empty()
+            }
+        } catch (Throwable throwable) {
```

Comment:
the try and catch can now be removed

Reply:
Wow, it literally took me 20 min to figure out what you meant 🤡 
Sure, I added the try-catch part based on https://github.com/tomtung/latex2unicode/blob/71979d6904fcf7efade786297983b881a1b019d4/src/main/scala/com/github/tomtung/latex2unicode/LaTeX2Unicode.scala#L143-L146
and assumed it needed to be there #scala-scared 😬

Reply:
So much prettier ✨ 😍